### DISCUSSION TITLE
Vite needs to import `@apollo/client/index.js` instead of `@apollo/client`

### DISCUSSION THREAD

**Discussion by mahmoudmoravej** [remix-run/remix#9264] [Q&A] [answered]
I started migrating from classic template to Vite, but I got these errors although the[ Apollo/Client](https://github.com/apollographql/apollo-client) package exports ESM too:
```
[vite] Named export 'gql' not found. The requested module '@apollo/client' is a CommonJS module
```
and 
```
TypeError: __vite_ssr_import_1__.useQuery is not a function
```

As https://github.com/apollographql/apollo-client/issues/9976#issuecomment-1768446694"@apollo/client/index.js" Adding `/index.js` to the end of the graphql imports solved the issue(why?)

But it is not a permanent fix as I have used `graphql-codegen` and running codegen again needs manual fixes. any permanent fix?

**Comment by viktorbasta** [comment, marked as answer]
@mahmoudmoravej  Adding this to **vite.config.ts** got rid of the error:
`
  ssr: {
    noExternal: ['@apollo/client'],
  }
`
